src/Spryker/Zed/EventBehavior/Business/Model/EventEntityTransferFilter.php

```diff
@@ -145,6 +149,93 @@ public function getEventTransfersAdditionalValues(array $eventTransfers, string
         return array_unique($additionalValues);
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\HydrateEventsRequestTransfer $hydrateEventsRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\HydrateEventsResponseTransfer
+     */
+    public function hydrateEventDataTransfer(HydrateEventsRequestTransfer $hydrateEventsRequestTransfer): HydrateEventsResponseTransfer
+    {
+        $hydrateEventsResponseTransfer = new HydrateEventsResponseTransfer();
+        $eventEntityTransfers = $hydrateEventsRequestTransfer->getEventEntities()->getArrayCopy();
+
+        if (count($eventEntityTransfers) === 0) {
+            return $hydrateEventsResponseTransfer;
+        }
+        usort($eventEntityTransfers, fn (EventEntityTransfer $a, EventEntityTransfer $b) => $a->getTimestamp() <=> $b->getTimestamp());
+
+        $hydrateEventsResponseTransfer->setIdTimestampMap($this->getIdsTimestampMap($eventEntityTransfers));
+
+        if ($hydrateEventsRequestTransfer->getAdditionalValueName()) {
+            $hydrateEventsResponseTransfer->setAdditionalValueTimestampMap($this->getAdditionalValueTimestampMap($eventEntityTransfers, $hydrateEventsRequestTransfer->getAdditionalValueName()));
+        }
+
+        if ($hydrateEventsRequestTransfer->getForeignKeyName()) {
+            $hydrateEventsResponseTransfer->setForeignKeyTimestampMap($this->getForeignKeyTimestampMap($eventEntityTransfers, $hydrateEventsRequestTransfer->getForeignKeyName()));
+        }
+
+        return $hydrateEventsResponseTransfer;
+    }
+
+    /**
+     * @param array<\Generated\Shared\Transfer\EventEntityTransfer> $eventTransfers
+     *
+     * @return array<int, int>
+     */
+    protected function getIdsTimestampMap(array $eventTransfers): array
+    {
+        $ids = [];
+        foreach ($eventTransfers as $eventTransfer) {
+            $ids[(int)$eventTransfer->getId()] = (int)$eventTransfer->getTimestamp();
```

Comment:
so, if we go with the easiest flow that we can check. E.g. I've updating some big form on backoffice that have more than 1 entity in it. I put new data in it and it will trigger n+1 different events. There are not a 0 chance that those events will have the same id. With this approach we are going to loose some of the events.

The same goes for other similar methods

Reply:
As you see, we use the last one for a specific entity. it's part of improvement